sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -184,23 +178,8 @@ class __SYCL_EXPORT SYCLMemObjT : public SYCLMemObjI {
       // ContiguousStorage. updateHostMemory works only with pointer to
       // continuous data.
       const size_t Size = MSizeInBytes / sizeof(DestinationValueT);
-      vector_class<DestinationValueT> ContiguousStorage(Size);
-      updateHostMemory(ContiguousStorage.data());
-      std::copy(ContiguousStorage.cbegin(), ContiguousStorage.cend(),
-                FinalData);
-    };
-  }
-
-  template <typename Destination>
-  __SYCL_DLL_LOCAL EnableIfOutputIteratorBool<Destination>
-  set_final_data(Destination FinalData) {
-    MUploadDataFunctor = [this, FinalData]() {
-      using DestinationValueT = iterator_value_type_t<Destination>;
-      // TODO if Destination is ContiguousIterator then don't create
-      // ContiguousStorage. updateHostMemory works only with pointer to
-      // continuous data.
```

Comment:
I believe this `TODO` is to be preserved as this is a point of improvement and we don't want to miss it.

Reply:
TODO is in the combined set_final_data method, 177-179 lines

Reply:
Oh, my bad. Didn't see it first.